gtm.go

```diff
@@ -276,6 +276,124 @@ func (b *BigIP) DeleteGTMAPoolMember(fullPathToAPool, serverFullPath, poolMember
+// GTMCNamePool contains information about each gtm/pool/cname.
+type GTMCNamePool struct {
+	Name                     string `json:"name,omitempty"`
+	Partition                string `json:"partition,omitempty"`
+	FullPath                 string `json:"fullPath,omitempty"`
+	Generation               int    `json:"generation,omitempty"`
+	AppService               string `json:"appService,omitempty"`
+	Description              string `json:"description,omitempty"`
+	Disabled                 bool   `json:"disabled,omitempty"`
+	DynamicRatio             string `json:"dynamicRatio,omitempty"`
+	Enabled                  bool   `json:"enabled,omitempty"`
+	FallbackMode             string `json:"fallbackMode,omitempty"`
+	LoadBalancingMode        string `json:"loadBalancingMode,omitempty"`
+	ManualResume             string `json:"manualResume,omitempty"`
+	TmPartition              string `json:"tmPartition,omitempty"`
+	QosHitRatio              int    `json:"qosHitRatio,omitempty"`
+	QosHops                  int    `json:"qosHops,omitempty"`
+	QosKilobytesSecond       int    `json:"qosKilobytesSecond,omitempty"`
+	QosLcs                   int    `json:"qosLcs,omitempty"`
+	QosPacketRate            int    `json:"qosPacketRate,omitempty"`
+	QosRtt                   int    `json:"qosRtt,omitempty"`
+	QosTopology              int    `json:"qosTopology,omitempty"`
+	QosVsCapacity            int    `json:"qosVsCapacity,omitempty"`
+	QosVsScore               int    `json:"qosVsScore,omitempty"`
+	TTL                      int    `json:"ttl,omitempty"`
+	VerifyMemberAvailability string `json:"verifyMemberAvailability,omitempty"`
+	MembersReference         struct {
+		Link            string `json:"link,omitempty"`
+		IsSubcollection bool   `json:"isSubcollection,omitempty"`
+	}
+}
+
+// GetGTMCNamePools returns a list of all Pool/CNAME records.
+func (b *BigIP) GetGTMCNamePools() (*GTMCNamePools, error) {
+	var p GTMCNamePools
+	err, _ := b.getForEntity(&p, uriGtm, uriPool, string(CNAMERecord))
+	if err != nil {
+		return nil, err
+	}
+
+	return &p, nil
+}
+
+// GetGTMCNamePool gets a Pool/CNAME by name.
+func (b *BigIP) GetGTMCNamePool(name string) (*GTMCNamePool, error) {
+	var w GTMCNamePool
+
+	err, ok := b.getForEntity(&w, uriGtm, uriPool, string(CNAMERecord), name)
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		return nil, nil
+	}
+
+	return &w, nil
+}
+
+// ********************************************************************************************************************
+// **************************************                            **************************************************
+// **************************************   GTM CNAME Pool Members   **************************************************
+// **************************************                            **************************************************
+// ********************************************************************************************************************
```

Comment:
Thank you for the commit! Would you mind removing the banner comments thought, as it will just keep the normal documentation consistent. Thank you!

Reply:
Done, thanks for checking out the pull request :)